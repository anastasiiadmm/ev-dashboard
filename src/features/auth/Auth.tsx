import bem from 'easy-bem';
import { Button, Col, Form, Row, Typography } from 'antd';
import { useState, ChangeEvent } from 'react';

import logo from '~/assets/images/logo.svg';
import FormField from '~/shared/ui/components/FormFIeld/FormField';
import '~/features/auth/Auth.scss';

const { Text } = Typography;

const Auth = () => {
  const b = bem('Auth');
  const [form] = Form.useForm();
  const [checked, setChecked] = useState(true);

  const onChangeCheckbox = (e: ChangeEvent<HTMLInputElement>) => {
    setChecked(e.target.checked);
  };

  const onFinish = async () => {};

  return (
    <Row justify='space-between' data-testid='auth-component' className={b()}>
      <div className={b('auth-block')}>
        <Col xs={{ span: 20, offset: 2 }} md={{ span: 24, offset: 2 }} lg={{ span: 22, offset: 1 }}>
          <div>
            <img src={logo} alt='logo' className={b('logo-image')} />
            <div>
              <Row className={b('buttons-row')}>
                <Text strong className={b('title')}>
                  Авторизация
                </Text>
                <Button type='link' className={b('register-button')}>
                  Забыли пароль?
                </Button>
              </Row>

              <Form
                form={form}
                initialValues={{ remember: true }}
                onFinish={onFinish}
                autoComplete='off'
                layout='vertical'
                size='large'
              >
                <FormField
                  data-testid='email_id_login'
                  type='email'
                  id='email_id'
                  name='username'
                  rules={[
                    {
                      required: true,
                      message: 'Заполните логин',
                    },
                  ]}
                  placeholder='Логин'
                />

                <FormField
                  data-testid='password_id'
                  type='password'
                  name='password'
                  placeholder='Пароль'
                />

                <FormField
                  className='checkbox-styles'
                  id='remember_id'
                  type='checkbox'
                  label={<span style={{ color: '#59647A' }}>Запомнить меня</span>}
                  valuePropName='password'
                  checked={checked}
                  onChange={onChangeCheckbox}
                />

                <Button disabled type='primary' htmlType='submit' className={b('login-button')}>
                  Войти
                </Button>
              </Form>
            </div>
          </div>
        </Col>
      </div>
    </Row>
  );
};

export default Auth;
