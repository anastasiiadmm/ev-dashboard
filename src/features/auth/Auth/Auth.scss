@use '../../../app/styles/variables' as variables;

.Auth {
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;

  &__auth-block {
    width: 400px;
    background-color: variables.$white-color;
    padding: 24px;
    border-radius: variables.$border-radius;
    border: 1px solid variables.$border-color;
  }

  &__buttons-row {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 20px 0;
  }

  &__title {
    font-size: 20px;
  }

  &__register-button {
    font-weight: variables.$font-600;
    font-size: 14px;
  }

  &__login-button {
    margin-top: 20px;
    width: 100%;
    font-size: 20px;
  }
}
