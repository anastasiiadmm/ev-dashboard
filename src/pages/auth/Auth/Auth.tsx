import bem from 'easy-bem';
import { Button, Col, Form, Row, Typography } from 'antd';
import React, { useState, ChangeEvent } from 'react';
import { observer } from 'mobx-react-lite';
import { useNavigate, Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import { logo } from '~/assets/images';
import { IUser } from '~/pages/auth/interfaces/IUser';
import { authStore } from '~/shared/api/store';
import { FormField } from '~/shared/ui';
import '~/pages/auth/Auth/Auth.scss';

const { Text } = Typography;

const Auth = observer(() => {
  const b = bem('Auth');
  const { t } = useTranslation();
  const navigate = useNavigate();
  const [form] = Form.useForm();
  const [checked, setChecked] = useState(true);

  const onChangeCheckbox = (e: ChangeEvent<HTMLInputElement>) => {
    setChecked(e.target.checked);
  };

  const onFinish = async (values: IUser) => {
    await authStore.loginUser(values);
    if (authStore.success) {
      navigate('/');
    }
  };

  return (
    <Row justify='space-between' data-testid='auth-component' className={b()}>
      <div className={b('auth-block')}>
        <Col xs={{ span: 20, offset: 2 }} md={{ span: 24, offset: 2 }} lg={{ span: 22, offset: 1 }}>
          <div>
            <img src={logo} alt='logo' className={b('logo-image')} />
            <div>
              <Row className={b('buttons-row')}>
                <Text strong className={b('title')}>
                  {t('login.sign_in')}
                </Text>
                <Link to='/reset-password'>
                  <Button type='link' className={b('register-button')}>
                    {t('login.forgot_password')}
                  </Button>
                </Link>
              </Row>

              <Form
                form={form}
                initialValues={{ remember: true }}
                onFinish={onFinish}
                autoComplete='off'
                layout='vertical'
                size='large'
              >
                <FormField
                  data-testid='email_id_login'
                  type='email'
                  id='email_id'
                  name='email'
                  placeholder={t('login.email')}
                  rules={[
                    {
                      type: 'email',
                      message: t('errors.the_input_is_not_valid_email'),
                    },
                    {
                      required: true,
                      message: t('errors.enter_your_email_address'),
                    },
                  ]}
                />

                <FormField
                  data-testid='password_id'
                  type='password'
                  name='password'
                  placeholder={t('login.your_password')}
                />

                <FormField
                  className='checkbox-styles'
                  id='remember_id'
                  type='checkbox'
                  label={<span style={{ color: '#59647A' }}>{t('login.remember_me')}</span>}
                  valuePropName='password'
                  checked={checked}
                  onChange={onChangeCheckbox}
                />

                <Button
                  loading={authStore.loading}
                  type='primary'
                  htmlType='submit'
                  className={b('login-button')}
                >
                  {t('login.sign_in_button')}
                </Button>
              </Form>
            </div>
          </div>
        </Col>
      </div>
    </Row>
  );
});

export default Auth;
