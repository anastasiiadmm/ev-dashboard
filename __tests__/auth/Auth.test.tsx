import React from 'react';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom";
import { BrowserRouter } from 'react-router-dom';

import Auth from "../../src/features/auth/Auth/Auth";

jest.mock('~/shared/utils/config', () => ({
  apiURL: 'http://localhost/8000',
}));

beforeAll(() => {
  process.env.NODE_ENV = 'test';
});

jest.mock('antd/lib/form/Form', () => {
  const actualForm = jest.requireActual('antd/lib/form/Form');
  return {
    ...actualForm,
    useForm: () => [{}, jest.fn()],
  };
});

jest.mock('../../src/features/auth/Auth/Auth', () => () => <div>Mocked Auth</div>);

describe('tests for Auth components',  () => {
  test('Render component toMatchSnapshot()', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Auth />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
