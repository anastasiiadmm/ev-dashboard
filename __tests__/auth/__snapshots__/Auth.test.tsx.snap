// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Auth components Render component toMatchSnapshot() 1`] = `
<DocumentFragment>
  <div>
    Mocked Auth
  </div>
</DocumentFragment>
`;
